src/components/TranslateDialog.vue

```diff
@@ -0,0 +1,194 @@
+			<p class="translate-dialog__message translate-dialog__message-translation">
+				{{ translatedMessage }}
+			</p>
+			<div class="translate-dialog__buttons">
+				<NcButton :disabled="loading"
+					class="translate-dialog__button"
+					@click="translateMessage">
+					<template v-if="loading" #icon>
+						<NcLoadingIcon />
+					</template>
+					{{ t('spreed', 'Translate') }}
+				</NcButton>
+			</div>
+		</div>
+	</NcModal>
+</template>
+
+<script>
+import { showError } from '@nextcloud/dialogs'
+
+import NcButton from '@nextcloud/vue/dist/Components/NcButton.js'
+import NcLoadingIcon from '@nextcloud/vue/dist/Components/NcLoadingIcon.js'
+import NcModal from '@nextcloud/vue/dist/Components/NcModal.js'
+import NcSelect from '@nextcloud/vue/dist/Components/NcSelect.js'
+
+import { translateText } from '../services/messagesService.js'
+
+export default {
+	name: 'TranslateDialog',
+
+	components: {
+		NcModal,
+		NcSelect,
+		NcButton,
+		NcLoadingIcon,
+	},
+
+	props: {
+		availableTranslations: {
+			type: Array,
+			default: () => [],
+		},
+
+		text: {
+			type: String,
+			required: true,
+		},
+	},
+
+	data() {
+		return {
+			translatedMessage: '',
+			availableTos: [],
+			selectedFrom: '',
+			selectedTo: '',
+			loading: false,
+		}
+	},
+
+	computed: {
+		availableFroms() {
+			return Array.from(new Set(this.availableTranslations.map(element => element.from)))
+		},
+	},
+
+	watch: {
+		selectedTo() {
+			this.translatedMessage = ''
+		},
+		selectedFrom() {
+			this.translatedMessage = ''
+		},
+	},
+
+	async mounted() {
+		// Set user language as "to"
+		const userLanguage = this.getUserLanguage()
+		if (this.hasFrom(userLanguage)) {
+			this.selectedTo = userLanguage
+		}
+
+		// FIXME: For some reason this does not return the auto detected language, let's address this in a follow up pr
+		const response = await translateText(this.text, '', this.selectedTo)
+		console.debug('response')
+	},
+	methods: {
+		getUserLanguage() {
+			return navigator.language.substring(0, 2)
+		},
+
+		hasFrom(from) {
+			return this.availableFroms.includes(from)
+		},
+
+		handleSelectFrom(from) {
+			this.availableTos = this.getAvailableTos(from)
+		},
+
+		getAvailableTos(from) {
+			return this.availableTranslations
+				.filter(element => element.from === from)
+				.map(element => element.to)
+		},
+
+		async translateMessage() {
+			try {
+				this.loading = true
+				const response = await translateText(this.text, this.selectedFrom, this.selectedTo)
+				this.loading = false
+				this.translatedMessage = response.data.ocs.data.text
+
+			} catch (error) {
+				this.loading = false
+				console.error(error)
+				showError(t('spreed', 'The message could not be translated'))
+			}
+
+		},
+	},
+}
+
+</script>
+
+<style lang="scss" scoped>
+.translate-dialog {
+	display: flex;
+	flex-direction: column;
+	min-height: 400px;
+	padding: calc(var(--default-grid-baseline) * 4);
+	&__message {
+		margin: calc(var(--default-grid-baseline) * 4);
+	}
```

Comment:
```suggestion
  &-translation {
    border: 1px solid var(--color-primary-element);
    border-radius: var(--border-radius-large);
    }
}
```

Maybe also wrap it with some border?